### ISSUE TITLE
The query string is built incorrectly when the path contains a parameter

### ISSUE THREAD

**Issue by simoncas** [liyasthomas/postwoman#87]
**Describe the bug**
The URL is crafted incorrectly when the path already contains a query string and when additional parameters are added through the options.
Example: https://reqres.in/api/users?param1=value1?param2=value2 

**To Reproduce**
url: http://whatever.com
path: /whatever?param1=value1
parameters: _param2_ as Key1 and _value2_ as Value1
Open the developer tools and select the network tab. Filter requests by XHR
Send the request by hitting the __Send__ button.
Using the developer tools,  analyze the HTTP request crafted by postwoman,  especially the URL

The URL is _https://reqres.in/api/users?param1=value1?param2=value2_ which isn't valid. The query string isn't valid since the parameters are preceded twice by a question mark. The second parameter should be preceded by a "&"  instead.

**Expected behavior**
The crafted URL should be https://reqres.in/api/users?param1=value1&param2=value2.
In other words, if the path already contains a query string then the additional parameters specified through the parameter section must be correctly added.

**Screenshots**
![Screenshot from 2019-08-28 18-36-24](https://user-images.githubusercontent.com/11412464/63874952-c952dc80-c9c2-11e9-9307-b739a547fe8c.png)

![Screenshot from 2019-08-28 18-34-09](https://user-images.githubusercontent.com/11412464/63874897-afb19500-c9c2-11e9-907b-a60ecd2bcbb7.png)

**Comment by liyasthomas** [comment]
While we're finding a fix to this issue, I suggest you to separate parameters from request `path` and use **Parameters** section to specify required parameters.

**Comment by simoncas** [comment]
I'm okay to work on this.
They are 2 ways to solve this issue.
* Check with a regex if the query string already contains parameters. Then based on the result, craft correctly the query string. **(quick)**

_or_ 

* Extract the parameters from the `path` and automatically fill the **parameters** sections. (postman style).  Keep in sync the parameters list and the **path**. **Sounds more like a feature**

**Comment by simoncas** [comment]
Sorry, I close this issue by inadvertence.

**Comment by liyasthomas** [comment]
Second option sounds great! I've assigned this to you. Looking forward on your PR :raised_hands:

**Comment by liyasthomas** [comment]
#62 can also be solved with this approach of binding key-value pairs to a fast URL entry input box. @NBTX

**Comment by AndrewBastin** [comment]
Yoo, I was kinda working on a PR myself for this... My idea to resolve this was on clicking the Send button, the path input is parsed and the values are appended to the params list.

**Comment by hosseinnedaee** [comment]
Hi
I pushed my changes [sync-path-and-parameters](https://github.com/hosseinnedaee/postwoman/tree/sync-path-and-parameters)
Could you please check the functionality. (I didn't clean the code yet)

**Comment by liyasthomas** [comment]
@hosseinnedaee I went through your approach. Seems to be working. Clean your code, optimize the logic. And send PR. I'll run few tests once again.

**Comment by hosseinnedaee** [comment]
@liyasthomas you're welcome
I have submitted a PR #152

**Comment by liyasthomas** [comment]
#152 solves this issue hence closing
